enumerables.rb

```diff
@@ -40,12 +75,24 @@ def my_map(&block)
       map_arr.push(block.call(i))
     end
     map_arr
+    to_enum
   end
 
-  def my_inject(thread, &block)
-    my_each do |item|
-      thread = block.call(thread, item)
+  def my_inject(accumulator = nil, operator = nill, &block)
+    if !block
+      if operator.nil?
+        operator = accumulator
+        acumulator = nil
+      end
+      operator.to_sym
+      each { |itm| accumulator = accumulator.nil? ? itm : accumulator.send(operator, itm) }
+    else
+      each { |itm| accumulator = accumulator.nil? ? itm : block.call(accumulator, itm) }
     end
-    thread
+    accumulator
+  end
```

Comment:
- [ ] enumerables #my_inject raises a "LocalJumpError" when no block or argument is given
      Failure/Error: expect { array.my_inject }.to raise_error(LocalJumpError)
      
        expected LocalJumpError, got #<NameError: undefined local variable or method `nill' for #<Array:0x00007fffe8fcc548>

- [ ]  enumerables #my_inject searches for the longest word in an array of strings
      Failure/Error: expect(words.my_inject(&search)).to eq words.inject(&search)
      
      NameError:
        undefined local variable or method `nill' for ["dog", "door", "rod", "blade"]:Array

- [ ]  enumerables #my_inject does not mutate the original array
      Failure/Error: array.my_inject { |num| num + 1 }
      
      NameError:
        undefined local variable or method `nill' for #<Array:0x00007fffe964df20>

- [ ]  enumerables #my_inject when a block is given without an initial value combines all elements of enum by applying a binary operation, specified by a block
      Failure/Error: expect(array.my_inject(&operation)).to eq array.inject(&operation)
      
      NameError:
        undefined local variable or method `nill' for #<Array:0x00007fffe9647918>

- [ ]  enumerables #my_inject when a block is given without an initial value combines all elements of enum by applying a binary operation, specified by a block::range
      Failure/Error: actual = range.my_inject { |prod, n| prod * n }
      
      NameError:
        undefined local variable or method `nill' for 5..50:Range

- [ ]  enumerables #my_inject when a block is given with an initial value combines all elements of enum by applying a binary operation, specified by a block
      Failure/Error: actual = range.my_inject(4) { |prod, n| prod * n }
      
      NameError:
        undefined local variable or method `nill' for 5..50:Range

- [ ] enumerables #my_inject when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method
      Failure/Error: expect(array.my_inject(:+)).to eq array.inject(:+)
      
      NameError:
        undefined local variable or method `nill' for #<Array:0x00007fffe963c388>
        Did you mean?  nil
                       nil?

- [ ]  enumerables #my_inject when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method:range
      Failure/Error: expect(range.my_inject(:*)).to eq range.inject(:*)
      
      NameError:
        undefined local variable or method `nill' for 5..50:Range